### PR TITLE
docs: api/grn_encoding: Move the to_string() and parse() reference to the header file

### PR DESCRIPTION
**Pull request by abetomo**
GitHub: GH-1817

Prepare to switch to documents automatically generated by Doxygen.

### CODE REVIEW

File: include/groonga/groonga.h [hunk 1]
```diff
@@ -158,12 +158,21 @@ GRN_API const char *
 grn_get_global_error_message(void);
 
 typedef enum {
+  /// The default encoding specified at build time (0).
+  /// The default at build time is "UTF-8".
   GRN_ENC_DEFAULT = 0,
+  /// "none" (1).
```
Comment (otegami) [inline include/groonga/groonga.h:164]:
```suggestion
  /// "none" (1)
```

File: include/groonga/groonga.h [hunk 2]
```diff
@@ -389,9 +398,26 @@ GRN_API size_t
 grn_get_memory_map_size(void);
 
 /* grn_encoding */
-
+/**
+ * \brief Return string representation for the encoding. For example,
+ *        `grn_encoding_to_string(GRN_ENC_UTF8)` returns `"utf8"`
```
Comment (otegami) [inline include/groonga/groonga.h:403]:
```suggestion
 *        `grn_encoding_to_string(GRN_ENC_UTF8)` returns `"utf8"`.
```

File: include/groonga/groonga.h [hunk 3]
```diff
@@ -389,9 +398,26 @@ GRN_API size_t
...
+ *
+ * \param encoding The encoding
+ *
+ * \return String representation for the encoding on success, "unknown"
+ *         on invalid encoding
```
Comment (otegami) [inline include/groonga/groonga.h:408]:
```suggestion
 *         on invalid encoding.
```

File: include/groonga/groonga.h [hunk 4]
```diff
@@ -389,9 +398,26 @@ GRN_API size_t
...
+ */
 GRN_API const char *
 grn_encoding_to_string(grn_encoding encoding);
+/**
+ * \brief Parse encoding name and returns \ref grn_encoding. For example,
+ *        `grn_encoding_parse("UTF8")` returns \ref GRN_ENC_UTF8
```
Comment (otegami) [inline include/groonga/groonga.h:414]:
```suggestion
 *        `grn_encoding_parse("UTF8")` returns \ref GRN_ENC_UTF8.
```

File: include/groonga/groonga.h [hunk 5]
```diff
@@ -389,9 +398,26 @@ GRN_API size_t
...
+ *
+ * \param name The encoding name
+ *
+ * \return \ref grn_encoding matching `name` on success, \ref GRN_ENC_UTF8 on
+ *         invalid encoding name
```
Comment (otegami) [inline include/groonga/groonga.h:419]:
```suggestion
 *         invalid encoding name.
```

**Review by otegami — APPROVED** [review]
The comments are minor, so I'll go ahead and mark it as LGTM for now 👍🏾

**Review by otegami — APPROVED** [review]
LGTM 👍🏾

File: include/groonga/groonga.h [hunk 6]
```diff
@@ -158,12 +158,21 @@ GRN_API const char *
 grn_get_global_error_message(void);
 
 typedef enum {
+  /// The default encoding specified at build time (0).
+  /// The default at build time is "UTF-8".
   GRN_ENC_DEFAULT = 0,
+  /// "none" (1)
+  /// Used for binary or as an internal initial value.
```
Comment (kou) [inline include/groonga/groonga.h:165]:
```suggestion
  /// Process strings as binary data (1)
```

File: include/groonga/groonga.h [hunk 7]
```diff
@@ -389,9 +398,26 @@ GRN_API size_t
 grn_get_memory_map_size(void);
 
 /* grn_encoding */
-
+/**
+ * \brief Return string representation for the encoding. For example,
+ *        `grn_encoding_to_string(GRN_ENC_UTF8)` return `"utf8"`.
```
Comment (kou) [inline include/groonga/groonga.h:403]:
```suggestion
 *        `grn_encoding_to_string(GRN_ENC_UTF8)` returns `"utf8"`.
```

File: include/groonga/groonga.h [hunk 8]
```diff
@@ -389,9 +398,26 @@ GRN_API size_t
...
+ * \param encoding The encoding
+ *
+ * \return String representation for the encoding on success, "unknown"
+ *         on invalid encoding.
+ */
 GRN_API const char *
 grn_encoding_to_string(grn_encoding encoding);
+/**
+ * \brief Parse encoding name and return \ref grn_encoding. For example,
+ *        `grn_encoding_parse("UTF8")` return \ref GRN_ENC_UTF8.
```
Comment (kou) [inline include/groonga/groonga.h:414]:
```suggestion
 *        `grn_encoding_parse("UTF8")` returns \ref GRN_ENC_UTF8.
```

**Pull request merged**